src/gemdat/plots/plotly/_shape.py

```diff
@@ -32,4 +34,131 @@ def shape(
     fig : plotly.graph_objects.Figure
         Output figure
     """
-    raise NotImplementedError
+    x_labels = ('X / Å', 'Y / Å', 'Z / Å')
+    y_labels = ('Y / Å', 'Z / Å', 'X / Å')
+
+    fig = make_subplots(
+        rows=2,
+        cols=3,
+        subplot_titles=x_labels,
+        shared_xaxes=True,
+        vertical_spacing=0.1,
+    )
+
+    distances = shape.distances()
+    distances_sq = distances**2
+
+    msd = np.mean(distances_sq)
+    std = np.std(distances_sq)
+    # Latex can not be rendered in 3D plotly plots: https://github.com/plotly/plotly.js/issues/608
+    title = f'{shape.name}: MSD = {msd:.3f} Å^2, std = {std:.3f}'
+
+    mean_dist = np.mean(distances)
+
+    _tmp_dict = defaultdict(list)
+    vector_dict = {}
+    if sites:
+        for site in sites:
+            _tmp_dict[site.label].append(site.coords)
+        for key, values in _tmp_dict.items():
+            vector_dict[key] = np.array(values) - shape.origin
+
+    coords = shape.coords
+
+    for col, (i, j) in enumerate(((0, 1), (1, 2), (2, 0))):
+        x_coords = coords[:, i]
+        y_coords = coords[:, j]
+
+        # Map of the coordinates
+        fig.add_trace(
+            go.Histogram2dContour(
+                x=x_coords,
+                y=y_coords,
+                name=y_labels[col],
+                showscale=False,
+                colorscale='Viridis',
+            ),
+            row=1,
+            col=col + 1,
+        )
+
+        # Highlight the (0,0,0) point
+        x0 = [0]
+        y0 = [0]
+        fig.add_trace(
+            go.Scatter(
+                x=x0,
+                y=y0,
+                mode='markers',
+                marker=dict(color='red'),
+                name='Sites ' + y_labels[col],
+                showlegend=False,
+            ),
+            row=1,
+            col=col + 1,
+        )
+        # Area around the zero
+        fig.add_shape(
+            type='circle',
+            xref='x',
+            yref='y',
+            x0=-mean_dist,
+            y0=-mean_dist,
+            x1=mean_dist,
+            y1=mean_dist,
+            line=dict(color='red', dash='dash'),
+            showlegend=False,
+            row=1,
+            col=col + 1,
+        )
+
+        for label, vects in vector_dict.items():
+            x_vs = vects[:, i]
+            y_vs = vects[:, j]
+
+            # Add shape points
+            fig.add_trace(
+                go.Scatter(
+                    x=x_vs,
+                    y=y_vs,
+                    mode='markers+text',
+                    text=[label] * len(x_vs),
+                    textposition='top center',
+                    textfont=dict(color='red'),
+                    marker=dict(color='red'),
```

Comment:
Cleaner and faster to use a dict literal, imo.

```suggestion
                    textfont={'color':'red'},
                    marker={'color':'red'},
```